### PR TITLE
Techdebt atualizar a pagina de documentacao do codex neodv 1517

### PR DESCRIPTION
**Pull request by henrique-lh**
## Description

With this pr:
- Revise all documentation; 
- Resolve the issue of code blocks not displaying in Sphinx.;
- Correct typographical and grammatical errors;
- Remove 'datarisk' as default parameters from different functions
- Adjust parameters, raises and returns from functions;
-  Improve readability (in my opinion);
- Fix conflict between return type annotation and actual return

## Related issues

- Closes: https://linear.app/datarisk/issue/NEODV-1517/[techdebt]-atualizar-a-pagina-de-documentacao-do-codex

## How to test it

Run sphinx in your machine and see the result

 
 
 
 **PR Summary by Typo**
------------

 **Summary**

This pull request includes various improvements to the documentation and codebase, such as updating links, renaming variables, and adding new features. Changes include:

1. Updating README links and renaming local variable names in docs.
2. Improving "tasource.rst" file by adding GCP support and changing code snippets.
3. Introducing a new way to retrieve a dataset using its DHash.
4. Changing import of environment variables from a code block to a `.env` file.
5. Allowing users to create a group with a description and disable/delete models.
6. Registering monitoring configurations for models using the `MLOpsPipeline.register_monitoring_config` method.
7. Adding support for running a preprocessing script before executing a model.

**Key Points**

1. Improved documentation and code consistency.
2. Added support for Google Cloud Platform (GCP) and new features.
3. Changed import of environment variables and added support for new clients.
4. Allowed users to create groups with descriptions and manage models.
5. Registered monitoring configurations for models using `MLOpsPipeline`.
6. Added support for running a preprocessing script before executing a model. 

 <h6>To turn off PR summary, please visit <a href="https://app.typoapp.io/settings/dev-analytics/notification?tab=codeHealth">Notification settings</a>.</h6>

### CODE REVIEW

File: src/mlops_codex/preprocessing.py [hunk 1]
```diff
@@ -953,21 +958,21 @@ def create(
...
         requirements_file : str
             Path of the requirements file. The packages versions must be fixed eg: pandas==1.0
-        schema : Union[str, dict]
-            Path to a JSON or XML file with a sample of the input for the entrypoint function. A dict with the sample input can be sending as well. Mandatory for Sync preprocessing
         group : str
-            Group the preprocessing is inserted. Default to 'datarisk' (public group)
-        extra_files : list, optional
+            Group the preprocessing is inserted.
+        schema : Optional[Union[str, dict]]
+            Path to a JSON or XML file with a sample of the input for the entrypoint function. A dict with the sample input can be sending as well. Mandatory for Sync preprocessing
```
Comment (64J0) [inline src/mlops_codex/preprocessing.py:964]: I'm not sure if this is specific for the sync script, but for the new async we're allowing parquet files too

File: src/mlops_codex/external_monitoring.py [hunk 2]
```diff
@@ -129,19 +140,30 @@ def upload_file(
...
+        requirements_file : Optional[str], optional
+            Path to your `requirements.txt` file. Defaults to None.
+        preprocess_file : Optional[str], optional
+            Path to your preprocessing file. Defaults to None.
+        preprocess_reference : Optional[str], optional
+            Preprocessing function entrypoint. Defaults to None.
+        shap_reference : Optional[str], optional
+            Shap function entrypoint. Defaults to None.
+        python_version : Optional[str], optional
+            Python version. Can be "3.8", "3.9", or "3.10". Defaults to "3.10".
```
Comment (64J0) [inline src/mlops_codex/external_monitoring.py:159]: I noticed that there's some inconsistency for the default python version. This is something we can improve for the codex. What you think @johnrc90 @henrique-lh?

File: src/mlops_codex/preprocessing.py [hunk 3]
```diff
@@ -953,21 +958,21 @@ def create(
...
+        extra_files : Optional[list], optional
             A optional list with additional files paths that should be uploaded. If the scoring function refer to this file they will be on the same folder as the source file
-        env : str, optional
+        env : Optional[str], optional
             Flag that choose which environment (dev, staging, production) of MLOps you are using. Default is True
-        python_version : str, optional
+        python_version : Optional[str], optional
             Python version for the preprocessing environment. Avaliable versions are 3.8, 3.9, 3.10. Defaults to '3.8'
```
Comment (64J0) [inline src/mlops_codex/preprocessing.py:970]: Here for example it defaults to 3.8

**Review by 64J0 — APPROVED** [review]
Other than the minor comments, this PR looks good to me. Great job @henrique-lh!

File: docs/source/connecting_datasource.rst [hunk 4]
```diff
@@ -15,37 +15,24 @@ To register your data source to MLOps, you must have the provider credentials ac
```
Comment (johnrc90) [inline docs/source/connecting_datasource.rst:15]: You need to reference this file in `docs/source/starters.rst` for it to show

File: docs/source/training_guide.rst [hunk 5]
```diff
@@ -134,9 +210,34 @@ For the AutoML we just need the data and the configuration parameters. You can c
...
+        "iterations": 1,
+        "metric": "ks",
+        "split_type": "random",
+        "stages": {
+            "models": ["logreg"]
+            }
+    }
+
+
+Running an AutoML training execution
```
Comment (johnrc90) [inline docs/source/training_guide.rst:234]: Saving a local training to Neomaril

File: src/mlops_codex/base.py [hunk 6]
```diff
@@ -48,7 +48,7 @@ def __init__(
         if url is None:
             url = os.getenv("MLOPS_URL")
         if url is None:
-            url = "https://neomaril.datarisk.net/"
+            url = "https://mlops.datarisk.net/"
```
Comment (johnrc90) [inline src/mlops_codex/base.py:51]: The backend is still neomaril.datarisk.net

File: src/mlops_codex/external_monitoring.py (hunk 2 above)
Comment (henrique-lh) [reply src/mlops_codex/external_monitoring.py:159]: Agree, I updated this, now the default version is Python 3.10 for every function. Maybe, we shouldn't even have a default value for this in my opinion

File: src/mlops_codex/base.py (hunk 6 above)
Comment (henrique-lh) [reply src/mlops_codex/base.py]: - https://github.com/datarisk-io/mlops_codex/pull/68/commits/e31e9dfa1604edcf0d27b76b1239df78ccfe37f7
- https://github.com/datarisk-io/mlops_codex/pull/68/commits/51ef22a6582845c2bdc90b429854f2b8b188e5ad
- https://github.com/datarisk-io/mlops_codex/pull/68/commits/872497ff6bb3d6fb0d6e6add57ae4d31b91c283a

File: docs/source/connecting_datasource.rst (hunk 4 above)
Comment (henrique-lh) [reply docs/source/connecting_datasource.rst:16]: https://github.com/datarisk-io/mlops_codex/pull/68/commits/143dbfaccb9bd83c20f413971eefb6ad122f3163

File: src/mlops_codex/preprocessing.py (hunk 3 above)
Comment (henrique-lh) [reply src/mlops_codex/preprocessing.py]: https://github.com/datarisk-io/mlops_codex/pull/68/commits/13337399b3e7eb948bd2e048488934f7eadfd110

File: src/mlops_codex/external_monitoring.py (hunk 2 above)
Comment (henrique-lh) [reply src/mlops_codex/external_monitoring.py:159]: https://github.com/datarisk-io/mlops_codex/pull/68/commits/13337399b3e7eb948bd2e048488934f7eadfd110

File: src/mlops_codex/preprocessing.py (hunk 1 above)
Comment (henrique-lh) [reply src/mlops_codex/preprocessing.py]: https://github.com/datarisk-io/mlops_codex/pull/68/commits/cba378e43e1d8e8edff6c5cf26d98246b96928c5

File: docs/source/training_guide.rst (hunk 5 above)
Comment (henrique-lh) [reply docs/source/training_guide.rst]: https://github.com/datarisk-io/mlops_codex/pull/68/commits/5d0442b7c9b8fe702477941f5e716c70e8ccf644

**Pull request merged**